From: Aurelien Jarno <aurelien@aurel32.net>
Date: Sun, 20 Jul 2014 19:16:31 +0200
Subject: MIPS: Loongson 3: Add Loongson LS3A RS780E 1-way machine definition
Forwarded: no

Add a Loongson LS3A RS780E 1-way machine definition, which only differs
from other Loongson 3 based machines by the UART base clock speed.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>
---
 arch/mips/include/asm/bootinfo.h      | 1 +
 arch/mips/loongson/common/machtype.c  | 1 +
 arch/mips/loongson/common/serial.c    | 1 +
 arch/mips/loongson/common/uart_base.c | 1 +
 4 files changed, 4 insertions(+)

Index: tree-jessie/arch/mips/include/asm/bootinfo.h
===================================================================
--- tree-jessie.orig/arch/mips/include/asm/bootinfo.h
+++ tree-jessie/arch/mips/include/asm/bootinfo.h
@@ -74,6 +74,7 @@ enum loongson_machine_type {
 	MACH_LEMOTE_A1101,
 	MACH_LEMOTE_A1201,
 	MACH_LEMOTE_A1205,
+	MACH_LOONGSON_3A780E1W,
 	MACH_LOONGSON_END
 };
 
Index: tree-jessie/arch/mips/loongson/common/machtype.c
===================================================================
--- tree-jessie.orig/arch/mips/loongson/common/machtype.c
+++ tree-jessie/arch/mips/loongson/common/machtype.c
@@ -31,6 +31,7 @@ static const char *system_types[] = {
 	[MACH_LEMOTE_A1101]		"lemote-3a-itx-a1101",
 	[MACH_LEMOTE_A1201]		"lemote-2gq-notebook-a1201",
 	[MACH_LEMOTE_A1205]		"lemote-2gq-aio-a1205",
+	[MACH_LOONGSON_3A780E1W]	"loongson-ls3a-rs780e-1w",
 	[MACH_LOONGSON_END]		NULL,
 };
 
Index: tree-jessie/arch/mips/loongson/common/serial.c
===================================================================
--- tree-jessie.orig/arch/mips/loongson/common/serial.c
+++ tree-jessie/arch/mips/loongson/common/serial.c
@@ -51,6 +51,7 @@ static struct plat_serial8250_port uart8
 	[MACH_LEMOTE_A1101]             {PORT_M(2, 25000000), {} },
 	[MACH_LEMOTE_A1201]             {PORT_M(2, 25000000), {} },
 	[MACH_LEMOTE_A1205]             {PORT_M(2, 25000000), {} },
+	[MACH_LOONGSON_3A780E1W]        {PORT_M(2, 33177600), {} },
 	[MACH_LOONGSON_END]		{},
 };
 
Index: tree-jessie/arch/mips/loongson/common/uart_base.c
===================================================================
--- tree-jessie.orig/arch/mips/loongson/common/uart_base.c
+++ tree-jessie/arch/mips/loongson/common/uart_base.c
@@ -42,6 +42,7 @@ void prom_init_loongson_uart_base(void)
 	case MACH_LEMOTE_A1101:
 	case MACH_LEMOTE_A1201:
 	case MACH_LEMOTE_A1205:
+	case MACH_LOONGSON_3A780E1W:
 		/* The CPU provided serial port (CPU) */
 		loongson_uart_base = LOONGSON_REG_BASE + 0x1e0;
 		break;
